File: 083.Remove_Duplicates_from_Sorted_List/Solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) { 
        if(head == null || head.next == null){
            return head;
        }
        //头结点和后一个时候相等
        if(head.val == head.next.val){
            //去掉头结点
            return deleteDuplicates(head.next);
        }else{
            //加上头结点
            head.next = deleteDuplicates(head.next);
            return head;
        }
    }
}

File: 083.Remove_Duplicates_from_Sorted_List/Solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def deleteDuplicates(self, head):
        if head is None:
            return None
        pos = head
        while pos is not None and pos.next is not None:
            if pos.val == pos.next.val:
                pos.next = pos.next.next
            else:
                pos = pos.next
        return head